core/src/test/scala/unit/kafka/coordinator/group/GroupMetadataManagerTest.scala

```diff
@@ -610,6 +610,42 @@ class GroupMetadataManagerTest {
   }
 
   @Test
+  def testLoadGroupsAndOffsetsFromDifferentSegments(): Unit = {
+    val groupMetadataTopicPartition = groupTopicPartition
+    val startOffset = 15L
+    val tp = new TopicPartition("foo", 0)
+
+
+    val logMock =  EasyMock.mock(classOf[Log])
+    EasyMock.expect(replicaManager.getLog(groupMetadataTopicPartition)).andStubReturn(Some(logMock))
+
+    val segment1Records = MemoryRecords.withRecords(startOffset, CompressionType.NONE,
+      createCommittedOffsetRecords(Map(tp -> 23L)) ++ Seq(buildStableGroupRecordWithMember("a")): _*)
+    val segment1End = expectGroupMetadataLoad(logMock, groupMetadataTopicPartition, startOffset, segment1Records)
+
+    val memberId = "b"
+    val committedOffsets = Map(tp -> 33L)
+    val segment2Records = MemoryRecords.withRecords(segment1End, CompressionType.NONE,
+      createCommittedOffsetRecords(committedOffsets) ++ Seq(buildStableGroupRecordWithMember(memberId)): _*)
+
+    val segment2End = expectGroupMetadataLoad(logMock, groupMetadataTopicPartition, segment1End, segment2Records)
+    EasyMock.expect(replicaManager.getLogEndOffset(groupMetadataTopicPartition)).andStubReturn(Some(segment2End))
+
+    EasyMock.replay(logMock, replicaManager)
+
+    groupMetadataManager.loadGroupsAndOffsets(groupMetadataTopicPartition, _ => ())
+
+    val group = groupMetadataManager.getGroup(groupId).getOrElse(fail("Group was not loaded into the cache"))
+    assertEquals(groupId, group.groupId)
+    assertEquals(Stable, group.currentState)
+    assertEquals(memberId, group.leaderId)
+    assertEquals(Set(memberId), group.allMembers)
+    assertEquals(1, group.allOffsets.size)
+    assertEquals(Some(33L), group.offset(tp).map(_.offset))
```

Comment:
is this a question to answer here or to be documented in code?
without the fix, another leader would be elected ("a"), the member Set is different(Set("a") and the offset is 23L